Content.Client/ContextMenu/UI/EntityMenuPresenter.cs

```diff
@@ -0,0 +1,421 @@
+            if (entity == null)
+                return;
+
+            // open verb menu?
+            if (args.Function == ContentKeyFunctions.OpenContextMenu)
+            {
+                _verbSystem.VerbMenu.OpenVerbMenu(entity);
+                args.Handle();
+                return;
+            }
+
+            // do examination?
+            if (args.Function == ContentKeyFunctions.ExamineEntity)
+            {
+                _systemManager.GetEntitySystem<ExamineSystem>().DoExamine(entity);
+                args.Handle();
+                return;
+            }
+
+            // do some other server-side interaction?
+            if (args.Function == EngineKeyFunctions.Use || args.Function == ContentKeyFunctions.AltActivateItemInWorld || args.Function == ContentKeyFunctions.Point ||
+                args.Function == ContentKeyFunctions.TryPullObject || args.Function == ContentKeyFunctions.MovePulledObject)
+            {
+                var inputSys = _systemManager.GetEntitySystem<InputSystem>();
+
+                var func = args.Function;
+                var funcId = _inputManager.NetworkBindMap.KeyFunctionID(func);
+
+                var message = new FullInputCmdMessage(_gameTiming.CurTick, _gameTiming.TickFraction, funcId,
+                    BoundKeyState.Down, entity.Transform.Coordinates, args.PointerLocation, entity.Uid);
+
+                var session = _playerManager.LocalPlayer?.Session;
+                if (session != null)
+                {
+                    inputSys.HandleInputCommand(session, func, message);
+                }
+
+                _verbSystem.CloseAllMenus();
+                args.Handle();
+                return;
+            }
+
+            if (_itemSlotManager.OnButtonPressed(args, entity))
+            {
+                _verbSystem.CloseAllMenus();
+            }
+        }
+
+        private bool HandleOpenContextMenu(in PointerInputCmdHandler.PointerInputCmdArgs args)
+        {
+            if (args.State != BoundKeyState.Down)
+                return false;
+
+            if (_stateManager.CurrentState is not GameScreenBase)
+                return false;
+
+            var player = _playerManager.LocalPlayer?.ControlledEntity;
+            if (player == null)
+                return false;
+
+            var coords = args.Coordinates.ToMap(_entityManager);
+
+            bool ignoreVisibility = _verbSystem.CanSeeAllContext || !_eyeManager.CurrentEye.DrawFov;
+
+            if (!_verbSystem.TryGetContextEntities(player, coords, out var entities, ignoreVisibility: ignoreVisibility))
+                return false;
+
+            // do we need to do container visibility checks? Note that while ghosts do ignore FOV, they cannot see into
+            // containers or access entities without sprites. Unless they should be allowed to? In that case add a `||
+            // !_eyeManager.CurrentEye.DrawFov` here
+            if (_verbSystem.CanSeeAllContext)
```

Comment:
ye why not just check ignoreVisibility here? i think it seems reasonable to let observers see into containers like that but maybe needs more discussion on discord

Reply:
honestly seems like verbsystem should be responsible for checking if it should be visible or not (in a separate public method), rather than having contextmenupresenter  handle getting eye manager getting sprite visiblity and stuff but idk

Reply:
Yeah looking at the code again, the visibility checks should really all be done in `VerbSystem.TryGetContextEntities`.
There is no good reason for `ContextMenuPresenter` to then comb through the results and remove some depending on the value of `_verbSystem.CanSeeAllContext`. I'll go fix that.

As to whether ghosts should be able to see through containers... maybe? But at the very least I think it would require a special check to see whether the container is a human, because otherwise it would make it very difficult for spectators to use the context menu around humans. Currently, if a spectator wants to right click a human, this is what they would see if they can see through containers:
![containers](https://user-images.githubusercontent.com/60421075/136139056-248736c5-edf3-4c38-b6e9-cdebbcba8e64.png)

Also, unless the scope of "CanSeeAll" is reduced, it probably should still be separated from what ghosts / not-FOV-restricted users can see. The `CanSeeAllContext` bool is currently set by the `/toggleallcontext` command and will lead to spawn-entities and map-entities being visible, in addition to ignoring containers:
![see_all](https://user-images.githubusercontent.com/60421075/136138671-f3f3227d-c519-4200-a592-4abdc70d09fc.png)
: